site/content/posts/2021-08-12-octant-0.23-release.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Dead container walking: making a case for Kubernetes Object Status "
+image: /img/posts/2021/08/18/octantStatus03.png
+excerpt: With release 0.23, Octant is a more accurate tool for troubleshooting scenarios
+author: David Espejo
+author_avatar: /img/contributors/david-espejo.jpg
+date: 2021-08-31
+categories: ['kubernetes']
+tags: ['octant', 'release', 'David Espejo']
+slug: octant-object-status
+---
+As an application developer, you may be familiar with kubectl and how to interpret the results you get from a very common command like:
+```
+    # kubectl get pods
+    # NAME                                      READY   STATUS    RESTARTS   AGE
+    3scale-kourier-control-54cc54cc58-px2dt   1/1     Running   3          4d2h
+    activator-67656dcbbb-hq62r                0/1     Running   161        4d2h
+    autoscaler-df6856b64-blxgm                1/1     Running   1          4d2h
+   ```
+But in case you are not, it could be challenging to accurately diagnose an issue when you see -according to the above example- that the "activator" Pod is in a Running *status* but 0 of its containers are Ready. 
+
+How could it be?
+
+It all comes to how the Kubernetes API handles Pod lifecycle
+
+# POD lifecycle details
+
+According to the [oficial docs](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/), the Kubernetes API has a PodStatus object with a field called **Phase** but there's also the container **State** field that Kubernetes tracks. The difference between the two lies in the level the API server is reffering to: the Pod or the container respectively.
+
+While there are five possible values for Pod phase:
+
+| Value | Description |
+| :----:  | ----- |
+| Pending | Configuration accepted by the server but not ready to run (e.g. init containers, pending image fetching, among others) |
+| Running | All of the Pod's containers created, at least one of them is running|
+| Succeeded | All containers successfully completed their task and then exited, especially true in the presence of OnFailure [restartPolicy](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) |
+| Failed | All containers terminated and at least one terminated in failure|
+| Unknown| Most likely the node is offline |
+
+There are also three container states defined in the Kubernetes API:
+
+| Value| Description|
+| ----| ----|
+|Waiting|   A task is pending to complete, for example a container image download or an init container still running|
+|Running| Container executing with no issues nor pending tasks|
+|Terminated| Container either ran to completion or it failed
+
+So, coming back to our example: what kubectl is showing for the "activator" Pod is the **Phase** field (in this case **Running** as the container has been created) but the application there it's not usable yet. How could we determine more accurately the actual situation of the Pod?
```

Comment:
Minor update: `but the application there it's not usable yet` -> `but the application there is not usable yet`